test/base.py

```diff
@@ -0,0 +1,46 @@
+import os
+import shutil
+import tempfile
+from pathlib import Path
+from unittest import TestCase
+
+
+class BaseTestCase(TestCase):
+    """Base class for our tests to house shared logic."""
+    _using_fake_hook_tools = False
+    _tmpdir = None
+
+    @property
+    def tmpdir(self):
+        if not self._tmpdir:
+            self._tmpdir = Path(tempfile.mkdtemp())
+            self.addCleanup(shutil.rmtree, self._tmpdir)
+        return self._tmpdir
+
+    def patch_env(self, **kwargs):
+        """Patch os.environ for a test, and ensure it gets restored.
+
+        Can be called multiple times.
+        """
+        old_env = dict(os.environ)
+        os.environ.update(kwargs)
+        self.addCleanup(os.environ.update, old_env)
```

Comment:
That'll add a new cleanup for every script created. A one time thing should be sufficient for us to clean the PATH and remove the temporary directory.